src/QsCompiler/SyntaxProcessor/SymbolTracker.fs

```diff
@@ -299,7 +297,7 @@ type SymbolTracker(globals: NamespaceManager, sourceFile, parent: QsQualifiedNam
             let argType, returnType =
                 decl.ArgumentType |> StripPositionInfo.Apply, decl.ReturnType |> StripPositionInfo.Apply
 
-            let idType = kind ((argType, returnType), decl.Information) |> ResolvedType.New
+            let idType = kind ((argType, returnType), decl.Information) |> ResolvedType.create qsSym.Range
```

Comment:
To clarify based on our other conversation, the PR didn't cause issues with IDE commands since it left alone the ranges in QsTypeParameter and UserDefinedType. However, having a range for types in multiple places was strange, so I updated the range in ResolvedType to distinguish between an inferred type and an annotated type, and marked the ranges for TPs and UDTs as obsolete.

Reply:
Currently, the IDE commands are still using the obsolete ranges instead of the new ResolvedType.Range, but I'd like to update that in another PR to avoid making this one too big.

Reply:
Ok, I am good with making it a separate PR, so I'll unblock the PR. However, please address the comment around removing other suppressions of deprecated warnings where possible (they are all small to update).